workspaces/libnpmexec/lib/index.js

```diff
@@ -86,20 +85,47 @@ const exec = async (opts) => {
     packages.push(args[0])
   }
 
+  // figure out whether we need to install stuff, or if local is fine
+  const localArb = new Arborist({
+    ...flatOptions,
+    path,
+  })
+  const localTree = await localArb.loadActual()
+
+  const getLocalManifest = ({ tree, name }) => {
+    // look up the package name in the current tree inventory,
+    // if it's found then return that normalized pkg data
+    let node
+    const nodes = tree.inventory.query('packageName', name)
+    for (const n of nodes) {
+      node = n
+      break
+    }
+
+    if (node) {
```

Comment:
instead of checking this after the fact, maybe a check on `nodes.size` before you even read a node from it is more clear?

Reply:
I'll stick with the deconstruction proposed in the previous comment and keep this check here 😊